flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/committer/FileCommitter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.file.sink.committer;
+
+import org.apache.flink.api.connector.sink.Committer;
+import org.apache.flink.connector.file.sink.FileSink;
+import org.apache.flink.connector.file.sink.FileSinkCommittable;
+import org.apache.flink.streaming.api.functions.sink.filesystem.BucketWriter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Committer implementation for {@link FileSink}.
+ */
+public class FileCommitter implements Committer<FileSinkCommittable> {
+	private static final Logger LOG = LoggerFactory.getLogger(FileCommitter.class);
+
+	private final BucketWriter<?, ?> bucketWriter;
+
+	public FileCommitter(BucketWriter<?, ?> bucketWriter) {
+		this.bucketWriter = checkNotNull(bucketWriter);
+	}
+
+	@Override
+	public List<FileSinkCommittable> commit(List<FileSinkCommittable> committables)  {
+		List<FileSinkCommittable> needRetry = new ArrayList<>();
+		for (FileSinkCommittable committable : committables) {
+			if (committable.hasPendingFile()) {
+				// We should always use commitAfterRecovery which contains additional checks.
+				try {
+					bucketWriter.recoverPendingFile(committable.getPendingFile()).commitAfterRecovery();
+				} catch (IOException e) {
+					LOG.error("Failed to commit {}", committable.getPendingFile());
+					needRetry.add(committable);
```

Comment:
Currently this will fail because the operators throw an exception. So for now we may have to simply return an empty list and open another JIRA as a reminder, to enable retry when the operators allow it. The other option would be to have a flag like `supportsRetry`, that is always `false` for now and in the future we enable it.

Reply:
Agree with that we should not use retry now, I changed to return the empty list for now.

Reply:
Agree with that we should not use retry now, I changed to return the empty list for now.